src/ser/mod.rs

```diff
@@ -96,12 +92,12 @@ impl<'a> Serializer<'a> {
 // which take 200+ bytes of ROM / Flash
 macro_rules! serialize_unsigned {
     ($self:ident, $N:expr, $v:expr) => {{
-        let mut buf: [u8; $N] = unsafe { super::uninitialized() };
+        let mut buf: [MaybeUninit<u8>; $N] = [MaybeUninit::uninit(); $N];
```

Comment:
I'm not sure the maybeuninit shenanigans are event worth the effort or unsafe below. The absolute largest buffer we're dealing with is a 20-byte buffer. What about just zero-initializing this (and below)?

Reply:
There will always be an intrinsic unsafe required with MaybeUninit.
I think it's ok and useful to use MaybeUninit here. And since it has been uninit for a long time (forever?), there would need to be a specific reason to remove it.